src/ml/neural_net/image_augmentation.cpp

```diff
@@ -171,8 +171,7 @@ image_augmenter::result float_array_image_augmenter::prepare_images(
   // Decode a batch of images to raw format (shared_float_arrays)
   // Also convert annotations and predictions per batch of images
   // to vectors of shared_float_arrays.
-  for (size_t i = 0; i < n; i++) {
-    const labeled_image& source = source_batch[i];
+  for (const labeled_image& source : source_batch) {
```

Comment:
Here is previous `n` is fixed to the predefined batch size,
we change it to the actual data batch size.

Reply:
I wonder this is a very simple logical error, why it is only revealed by now? I should fail the unit test very easily.